pkg/operators/klusterlet/controllers/klusterletcontroller/klusterlet_controller.go

```diff
@@ -209,8 +209,8 @@ func (n *klusterletController) sync(ctx context.Context, controllerContext facto
 		appliedManifestWorkClient: n.appliedManifestWorkClient,
 	}
 
-	if config.InstallMode == operatorapiv1.InstallModeDetached {
-		managedClusterClients, err = n.buildManagedClusterClientsDetachedMode(ctx, n.kubeClient, config.KlusterletNamespace, config.ExternalManagedKubeConfigSecret)
+	if config.InstallMode == operatorapiv1.InstallModeDetached || config.InstallMode == operatorapiv1.InstallModeHosted {
```

Comment:
I am thinking can we just change the installMode value to `Hosted` when it is `Detached` here, so we can remove `Detached` at other places？ The advantage of this is to cancel Detached in the future just need to change here.

Reply:
That's a good point!  Updated, now we replace the mode to hosted at the first place if the mode is detached.